File: workspace/pset6/Sentimental/vigenere/vigenere.py
from cs50 import get_string
import sys

def main():
    if(len(sys.argv) != 2):
        print("Error: Expected only ONE KEY STRING parameter")

    keyword = sys.argv[1]
    i = 0
    while(i < len(keyword)):
        if(((keyword[i] < 'a' and keyword[i] > 'z') or (keyword[i] < 'A' and keyword[i] > 'Z'))):
            print("Error: Expected only alphabets in keyword.")
        i = i + 1

    string = get_string("Plaintext: ")
    msg = list(string)
    newMsg = modifymsg(msg, keyword)
    print("Ciphertext: ", end = "")
    print(''.join(newMsg))



def modifymsg(msg, keyword):
    j = 0
    i = 0
    ciphertext = []
    while(i < len(msg)):
        if((msg[i] >= 'A' and msg[i] <= 'Z') or (msg[i] >= 'a' and msg[i] <= 'z')):
            if(isLower(keyword[j])):
                key = ord(keyword[j]) - 97
            else:
                key = ord(keyword[j]) - 65

            if(isLower(msg[i]) == True):
                val = 97
            else:
                val = 65
            newChar = (ord(msg[i]) - val + key) % 26 + val
            ciphertext.append(chr(newChar))
            j = j + 1
            if(j > len(keyword) - 1):
                j = 0

        i = i + 1

    return ciphertext


def isLower(c):
    if(c >= 'a'and c <= 'z'):
        return True
    else:
        return False



if __name__ == "__main__":
    main()

File: workspace/pset2/vigenere/vigenere.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

void modifymsg(string msg, string keyword);
bool isLower(char c);

int main(int argc, string argv[]){
    if(argc != 2){
        printf("Error: Expected only 1 keyword.\n");
        return 1;
    }
    string keyword = argv[1];
    int i = 0;
    while(keyword[i] != '\0'){
        if(!((keyword[i] >= 'a' && keyword[i] <= 'z') || (keyword[i] >= 'A' && keyword[i] <= 'Z'))){
            printf("Error: Expected only alphabets in keyword.\n");
            return 1;
        }
        i++;
    }
    i = 0;
    string msg = get_string("plaintext: ");

    modifymsg(msg, keyword);
    printf("ciphertext: ");
    while(msg[i] != '\0'){
        printf("%c", msg[i]);
        i++;
    }
    printf("\n");
}

void modifymsg(string msg, string keyword){
    printf("msg = %s\n", msg);
    printf("keyword = %s\n", keyword);
    int j = 0;
    int i = 0;
    while(msg[i] != '\0'){
        if((msg[i] >= 'A' && msg[i] <= 'Z') || (msg[i] >= 'a' && msg[i] <= 'z')){
            if(keyword[j] == '\0'){
                j = 0;
            }

            int k = 0;
            if(isLower(keyword[j])){
                k = keyword[j] - 97;
            } else{
                k = keyword[j] - 65;
            }

            if(isLower(msg[i])){
                msg[i] = (msg[i] - 97 + k) % 26 + 97;
            } else {
                msg[i] = (msg[i] - 65 + k) % 26 + 65;
            }
            j++;
        }
        i++;
    }
}

bool isLower(char c){
    if(c >= 'a' && c <= 'z'){
        return true;
    }
    return false;
}